numba/dispatcher.py

```diff
@@ -762,16 +763,51 @@ def compile(self, sig):
 
             self._pre_compile(args, return_type, flags)
 
-            # Clone IR to avoid mutation in rewrite pass
-            cloned_func_ir = self.func_ir.copy()
-            cres = compiler.compile_ir(typingctx=self.typingctx,
+            # This code is pessimistic and costly, but it is a not often trodden
```

Comment:
Why are these not done inside the compiler pipeline?

Reply:
As in push this down into `compiler.compile_ir` or deeper, itself branching on `self.lifted_from is not None`? I was trying to keep this out of the pipeline itself as it's applicable solely to lifted loops, but I suppose also that if a lifted loop compilation is required outside the context of this dispatcher then the problems that this fixes don't get fixed. I'll try pushing it down and see what happens.

Reply:
I've moved this into `compiler.compile_ir`, pushing it into the pipeline itself would lead to having to maintain two compile results in the pipeline state and a load of invasive stuff to do so.